packages/synapse-react-client/src/components/SynapseTable/SynapseTable.integration.test.tsx

```diff
@@ -0,0 +1,611 @@
+      {...queryWrapperPropOverrides}
+    >
+      <QueryVisualizationWrapper>
+        <QueryContextConsumer>
+          {context => {
+            // Capture the query context so we can use it in our tests
+            queryContext = context
+            return <></>
+          }}
+        </QueryContextConsumer>
+        <SynapseTable {...props} />
+      </QueryVisualizationWrapper>
+    </QueryWrapper>,
+    {
+      wrapper: createWrapper(),
+    },
+  )
+}
+
+jest.spyOn(HasAccessModule, 'HasAccessV2').mockImplementation(() => {
+  return <div data-testid="HasAccess"></div>
+})
+
+jest.spyOn(UserCardModule, 'default').mockImplementation(() => {
+  return <div data-testid="UserCard"></div>
+})
+
+jest.spyOn(AddToDownloadListV2Module, 'default').mockImplementation(() => {
+  return <div data-testid="AddToDownloadListV2" />
+})
+
+describe('SynapseTable tests', () => {
+  beforeAll(() => {
+    server.listen()
+    server.use(
+      ...getHandlersForTableQuery(queryResultBundle),
+
+      rest.post(
+        `${getEndpoint(BackendDestinationEnum.REPO_ENDPOINT)}${ENTITY_HEADERS}`,
+        async (req, res, ctx) => {
+          const requestBody: ReferenceList = (await req.json())
+            .references as ReferenceList
+          const responseBody: PaginatedResults<EntityHeader> = {
+            results: requestBody.map((reference: Reference, index) => {
+              return {
+                id: reference.targetId,
+                name: `mock entity ${index}`,
+                type: 'org.sagebionetworks.repo.model.FileEntity',
+                benefactorId: mockProjectIds[0],
+                createdOn: '2023-03-31T18:30:00.000Z',
+                modifiedOn: '2023-03-31T18:30:00.000Z',
+                createdBy: MOCK_USER_ID.toString(),
+                modifiedBy: MOCK_USER_ID.toString(),
+                isLatestVersion: true,
+              }
+            }),
+          }
+          return res(ctx.status(200), ctx.json(responseBody))
+        },
+      ),
+      rest.get(
+        `${getEndpoint(
+          BackendDestinationEnum.REPO_ENDPOINT,
+        )}${ENTITY_ID_VERSION(':id', ':version')}`,
+        async (req, res, ctx) => {
+          const responseBody: FileEntity = {
+            id: `${req.params.id!}`,
+            name: `Mock Entity with Id ${req.params.id}`,
+            versionNumber: parseInt(req.params.version as string),
+            versionLabel: `v${req.params.version}`,
+            versionComment: 'test',
+            modifiedOn: '2021-03-31T18:30:00.000Z',
+            modifiedBy: MOCK_USER_ID.toString(),
+            dataFileHandleId: mockFileHandle.id,
+            etag: 'etag',
+            concreteType: 'org.sagebionetworks.repo.model.FileEntity',
+          }
+          return res(ctx.status(200), ctx.json(responseBody))
+        },
+      ),
+    )
+  })
+  afterEach(() => server.restoreHandlers())
+  afterAll(() => server.close())
+
+  describe('Properly renders supplemental UI columns', () => {
```

Comment:
Not sure if I should parameterize this with `.each` and pass along the table type and expected values, or have duplicate code. Thoughts?